absorption_demo.ipynb

```diff
@@ -0,0 +1,261 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Testing Mantid's CylindricalAbsorption wrapper\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Information:</b> This notebook shows how to use Mantid's CylindricalAbsorption algorithm with a `scipp` Dataset.\n",
+    "\n",
+    "<b>Requirements:</b> To run this notebook, you need `mantid`, `scipp` and `numpy` installed as well as the Python script `absorption.py` placed in the same folder as this notebook.   \n",
+    "Additionally, the test file `wish_test.nxs` should reside in the same directory.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import scipp as sc\n",
+    "from scipp import Dim\n",
+    "import mantid.simpleapi as simpleapi\n",
+    "from scipp.compat.mantid import load\n",
+    "from absorption import absorption_correction\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading data\n",
+    "\n",
+    "Load a suitable dataset first. This is a **HUGE** file so each operation will be quite slow.\n",
+    "The load operation takes ~8s on the DMSC workstation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filename = \"wish_test.nxs\"\n",
+    "dataset = sc.compat.mantid.load(filename=filename)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The file is loaded as a `DataArray` with following sizes:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(dataset.dims)\n",
+    "print(dataset.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining absorption parameters\n",
+    "\n",
+    "Our absorption should be calculated for the sample."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scatter_from = \"Sample\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Calculating absorption\n",
+    "\n",
+    "Now we are ready to calculate the correction.\n",
+    "This calculation takes some time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "correction = absorption_correction(dataset, ScatterFrom=scatter_from)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The correction dataset uses wavelengths units, so we need to convert it back to time-of-flight, as in the original sample."
```

Comment:
Do we? Isn't the absorption applied while the data is in units of wavelength?

Reply:
it is, I guess. Removed the conversion